pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PublicSettings.scala

```diff
@@ -1,76 +1,50 @@
 package com.gu.pandomainauth
 
-import java.util.concurrent.atomic.AtomicReference
-import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
 import com.amazonaws.services.s3.AmazonS3
-import com.gu.pandomainauth.PublicSettings.validateAndParseKeyText
-import com.gu.pandomainauth.service.Crypto
-import org.slf4j.LoggerFactory
+import com.gu.pandomainauth.Settings.Loader
+import com.gu.pandomainauth.SettingsFailure.SettingsResult
+import com.gu.pandomainauth.service.CryptoConf
 
 import java.security.PublicKey
-import java.util.regex.Pattern
-import scala.concurrent.ExecutionContext
+import java.util.concurrent.Executors.newScheduledThreadPool
+import java.util.concurrent.{Executors, ScheduledExecutorService}
 import scala.concurrent.duration._
 
 /**
  * Class that contains the static public settings and includes mechanism for fetching the public key. Once you have an
  * instance, call the `start()` method to load the public data.
-  *
-  * @param settingsFileKey the settings file for the domain in the S3 bucket (eg local.dev.gutools.co.uk.public.settings)
-  * @param bucketName      the name of the S3 bucket (eg pan-domain-auth-settings)
-  * @param s3Client        the AWS S3 client that will be used to download the settings from the bucket
-  * @param scheduler       optional scheduler that will be used to run the code that updates the bucket
  */
-class PublicSettings(settingsFileKey: String, bucketName: String, s3Client: AmazonS3,
-                     scheduler: ScheduledExecutorService = Executors.newScheduledThreadPool(1)) {
+class PublicSettings(loader: Settings.Loader, scheduler: ScheduledExecutorService) {
 
-  private val agent = new AtomicReference[Option[PublicKey]](None)
```

Comment:
I'm sure this change is legit, but I don't think I'm clear in my own mind about how either:
1. `AtomicReference` is not needed here, or else
2. it's removal has been compensated for elsewhere

Reply:
ah, I think I've tracked it down -- the store has been moved to the shared `Settings` class https://github.com/guardian/pan-domain-authentication/pull/151/files#diff-91c51062f921c3d5c266a62c85f90eb5e6afce3eeeea0dbf299e9a7f8320988c

Reply:
That's right! The code is now in `Settings.Refresher`, here:

https://github.com/guardian/pan-domain-authentication/blob/c4835010a87c1ae854c3e8ca4dadf086b5ceb6a4/pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/Settings.scala#L82-L85